### PR TITLE
FreeBSD: cfmakesane, EVFILT_*

### PR DESCRIPTION
**Pull request by myfreeweb**
Depends on: https://github.com/rust-lang/libc/pull/887

### CODE REVIEW

File: src/sys/event.rs [hunk 1]
```diff
@@ -39,6 +39,8 @@ libc_enum! {
     #[cfg_attr(not(target_os = "netbsd"), repr(i16))]
     pub enum EventFilter {
         EVFILT_AIO,
+        #[cfg(target_os = "freebsd")]
```
Comment (Susurrus) [inline src/sys/event.rs]: Each one of these need doccomments describing them.

**Comment by asomers** [comment]
Thanks for your submission!  Have you checked the other BSDs to see how many define these symbols too?  I see that `cfmakesane` is in Dragonfly, but I haven't checked any of the others.

**Comment by myfreeweb** [comment]
@asomers the `libc` crate only exposes it on FreeBSD… I'll add it there first.

@Susurrus added.

**Comment by myfreeweb** [comment]
`cfmakesane` was added to DragonFly [last December](http://gitweb.dragonflybsd.org/dragonfly.git/commitdiff/81c399c9190a95858ba3279d24309d944b124736), not in any release yet. It was in FreeBSD for [seven years](https://github.com/freebsd/freebsd/commit/568dc19750ccec553ab3df463967352da528cd93).

**Comment by asomers** [comment]
Ok.  What about  `EVFILT_EMPTY` ?

**Comment by myfreeweb** [comment]
I have documented `_EMPTY`. There's nothing in the manpage for `_SENDFILE`.

(hmm, none of the other values have been documented here before…)

**Comment by Susurrus** [comment]
@myfreeweb Yes, there has been a severe lack of documentation within `nix` in the past, so we're trying to fix that as we go. It'd be much appreciated if you could add doccomments for the other constants that are there since you're already digging through these docs.

**Comment by myfreeweb** [comment]
Yeah I'll write some docs there soon.

\+ one more addition: fix for `write_int` ioctls. On FBSD & DFly they're defined with an [`IOWINT` macro](https://github.com/freebsd/freebsd/blob/6fceb757ffd5b050eca4c75ecbd350710b11db5e/sys/sys/ioccom.h#L59) that actually uses the `VOID` type instead of `IN`, so wrong numbers were generated.

**Comment by Susurrus** [comment]
I'm currently rewriting the macro logic because of the conversation in #824 and #781, so let's skip the ioctl stuff for this PR. I should have my rewrite done by this weekend and I'll CC you on the PR to review that it corrects things for the BSDs as well.

File: src/sys/event.rs [hunk 2]
```diff
@@ -51,7 +54,15 @@ libc_enum! {
         #[cfg(any(target_os = "ios", target_os = "macos"))]
         EVFILT_MACHPORT,
         EVFILT_PROC,
+        /// Returns events associated with the process referenced by a given
+        /// process descriptor, created by pdfork(2). The events to monitor are:
```
Comment (Susurrus) [inline src/sys/event.rs]: Backticks needed around `pdfork()` (leave out the "2").

File: src/sys/event.rs [hunk 3]
```diff
@@ -51,7 +54,15 @@ libc_enum! {
...
+        /// 
+        /// - NOTE_EXIT: the process has exited. The exit status will be stored in data.
+        #[cfg(target_os = "freebsd")]
+        EVFILT_PROCDESC,
         EVFILT_READ,
+        #[cfg(target_os = "freebsd")]
+        EVFILT_SENDFILE,
```
Comment (Susurrus) [inline src/sys/event.rs]: @asomers Could you help us dig up some docs for this constant?

File: src/sys/ioctl/mod.rs [hunk 4]
```diff
@@ -325,7 +325,7 @@ macro_rules! ioctl {
         pub unsafe fn $name(fd: $crate::libc::c_int,
                             data: $crate::libc::c_int)
                             -> $crate::Result<$crate::libc::c_int> {
-            convert_ioctl_res!($crate::libc::ioctl(fd, iow!($ioty, $nr, ::std::mem::size_of::<$crate::libc::c_int>()) as $crate::sys::ioctl::ioctl_num_type, data))
+            convert_ioctl_res!($crate::libc::ioctl(fd, iowint!($ioty, $nr, ::std::mem::size_of::<$crate::libc::c_int>()) as $crate::sys::ioctl::ioctl_num_type, data))
```
Comment (Susurrus) [inline src/sys/ioctl/mod.rs]: Let's remove the ioctl changes from this PR, please comment in #833 with a brief description and some test cases that fail on FreeBSD (see `test/sys/test_ioctl.rs` for existing test cases and please confirm which ones fail). @asomers could you help out here as well?

File: src/sys/termios.rs [hunk 5]
```diff
@@ -793,6 +793,18 @@ pub fn cfmakeraw(termios: &mut Termios) {
     termios.update_wrapper();
 }
 
+/// Configures the port to "sane" mode (like the configuration of a newly created terminal).
+///
+/// Note that this is a non-standard function, available on FreeBSD.
+#[cfg(target_os = "freebsd")]
+pub fn cfmakesane(termios: &mut Termios) {
```
Comment (Susurrus) [inline src/sys/termios.rs]: This needs a link to the man pages.

File: src/sys/event.rs (hunk 3 above)
Comment (asomers) [reply src/sys/event.rs]: How about "indicates the completion of an asynchronous `sendfile(2)` call".  I'm afraid be more specific than that.

**Comment by myfreeweb** [comment]
Removed the ioctl fix here, updated comments.

rust-lang/libc#887 has landed, but not in a release yet.

**Comment by Susurrus** [comment]
Please move the CHANGELOG entries into the `UNRELEASED` section, as they weren't a part of the 0.10.0 release. Then it's LGTM.

**Comment by myfreeweb** [comment]
done

**Comment by Susurrus** [comment]
Looks like it'll need another rebase to deal with changes to the CHANGELOG file. When you do please post back here, as I don't get an email on changes to PRs that aren't comments.

**Comment by myfreeweb** [comment]
rebased

**Comment by Susurrus** [comment]
Thanks for your contribution!

bors r+

**Pull request merged**